desktop/core/src/desktop/models.py

```diff
@@ -1855,9 +1855,11 @@ def get_main_quick_action(self, apps):
       LOG.exception('Could not load back default app')
 
     if default_interpreter:
-      return default_interpreter[0]
-    else:
-      return default_app
+      if DEFAULT_INTERPRETER.get() != '':
+        for di in default_interpreter:
+          if di['name'] == DEFAULT_INTERPRETER.get():
+            return di
+    return default_interpreter[0]
```

Comment:
Nit: Maybe for inner if
```python
else:
  return default_interpreter[0]
```

Reply:
It should only return if it finds it otherwise it'll fallback to the return on line 1862.